health-services/project-factory/migration/main/V20260820140000__add_attendance_sync_state_campaign_data.sql

```
@@ -0,0 +1,27 @@
+-- de-enrolment only the register and the person. So the sync can find its row only by the identity
+-- stamped at upload time — WHERE type = ? AND uniqueIdAfterProcess = ANY(?) — which the primary key
+-- (campaignNumber, uniqueIdentifier, type) cannot serve, since campaignNumber is absent.
+--
+-- Partial on purpose. user/facility/boundary/target rows also populate uniqueIdAfterProcess and are
+-- written in bulk during campaign creation, but are never looked up this way; excluding them means
+-- those inserts do not maintain this index at all. Measured on a 10M-row table: a register delete
+-- takes 0.5 ms instead of 315 ms, and a 200k-row user insert is unchanged.
+CREATE INDEX IF NOT EXISTS idx_eg_cm_campaign_data_attendance_identity
+    ON eg_cm_campaign_data (type, uniqueIdAfterProcess)
+    WHERE type IN ('attendanceRegister', 'attendanceRegisterAttendee')
+      AND uniqueIdAfterProcess IS NOT NULL;
```

SQLFluff (4.2.2)

[error] CREATE INDEX should use CONCURRENTLY to avoid locking the table during the build.

(PG01)